sklearn/preprocessing/tests/test_encoders.py

```diff
@@ -413,6 +413,32 @@ def test_one_hot_encoder_inverse():
         assert_raises_regex(ValueError, msg, enc.inverse_transform, X_tr)
 
 
+@pytest.mark.parametrize("X", [
+    [1, 2],
+    np.array([3., 4.])
+    ])
+def test_X_is_not_1D(X):
+    oh = OneHotEncoder()
+
+    msg = ("Expected 2D array, got 1D array instead")
+    with pytest.raises(ValueError, match=msg):
+        oh.fit(X)
+    with pytest.raises(ValueError, match=msg):
+        oh.fit_transform(X)
+
+
+def test_X_is_not_1D_pandas():
+    pd = pytest.importorskip('pandas')
```

Comment:
This can be combined with the above using `pytest.mark` syntax like https://docs.pytest.org/en/latest/skipping.html#skip-xfail-with-parametrize... (It might be best if we implement a requires_pandas mark in the long term?)

Reply:
@jnothman I think this one is not that easy to do now, as 1) I am not sure the `pytest.importorskip` functions as a mark like that, but 2) we also still need the returned `pd` of it, so we should still do a pandas import inside the test.